packages/controller/src/resources/tenants/index.ts

```diff
@@ -15,10 +15,53 @@
  */
 
 import { KubeConfig } from "@kubernetes/client-node";
-import { ResourceCollection, Namespace } from "@opstrace/kubernetes";
+import { ResourceCollection, Namespace, Secret } from "@opstrace/kubernetes";
 import { State } from "../../reducer";
 import { getTenantNamespace } from "../../helpers";
 
+const CERT_SECRET_NAMESPACE = "ingress"
+const CERT_SECRET_NAME = "https-cert"
+
+const getCertSecretCopy = (targetNameSpace: string,  state: State, kubeConfig: KubeConfig) => {
+  const certSecretSource = state.kubernetes.cluster.Secrets.resources.find(
+    secret =>
+      secret.name === CERT_SECRET_NAME &&
+      secret.namespace === CERT_SECRET_NAMESPACE
+  );
+
+  if (!certSecretSource) {
+    // TODO: Is this a situation that needs to be handled, and if so, how?
+    throw new Error()
+  }
+
+  const certSecretSourceTenant = new Secret(
+    {
+      apiVersion: "v1",
+      data: {
+        ...certSecretSource.data,
+      },
+      kind: "Secret",
+      metadata: {
+        name: "https-cert",
+        namespace: targetNameSpace,
+        annotations: {
+          // select the namespaces that have a label named `tenant`
+          // TODO: is this still needed?
+          "kubed.appscode.com/sync": "tenant"
```

Comment:
as the secret is not propagated by cubed anymore, is this annotation still needed?

Reply:
Yes, remove everything related to kubed. You can do it in one separate commit if you prefer.